### PR TITLE
Nextjs starter project

### PR DESCRIPTION
**Pull request by orlyohreally**
Trello card: https://trello.com/c/ggWpUvQY/1329-nextjs-starter-project

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ import Butter from 'buttercms';
 
 let butter;
 
-const preview = process.env.PREVIEW === "true"
+const previewSetting = process.env.PREVIEW
+// make preview mode by default
+const preview = previewSetting === "true" || previewSetting === undefined
```
Comment (jakelumetta) [inline lib/api.js:7]: Let's just rename `.env.sample` to `.env` and set `PREVIEW=true` as a default in it.  Like this:

```
PREVIEW=true
# Set your NEXT_PUBLIC_BUTTER_CMS_API_KEY in this file. For example NEXT_PUBLIC_BUTTER_CMS_API_KEY =your_token
# Get your free API token https://buttercms.com/join/
```

Then I think we can remove this line 7.

Also then when someone runs the step 2 command of `$ echo 'NEXT_PUBLIC_BUTTER_CMS_API_KEY =<Your API Token>' >> .env`, it will then insert the env variable at the bottom of the `.env` file

File: pages/_document.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Html, Head, Main, NextScript } from 'next/document'
```
Comment (jakelumetta) [inline pages/_document.js:1]: Why is this file needed?

File: README.md [hunk 3]
```diff
@@ -3,77 +3,64 @@ Live Demo: https://nextjs-starter-buttercms.vercel.app/
```
Comment (jakelumetta) [inline README.md:3]: Given that `npm run build` and `npm run start` seem to be more of a sanity/safety check for the developer that their application _will_ deploy to Vercel successfully, we don't need to include them in the README. We should instead ensure our starter app does indeed deploy fine to Vercel and then devs won't need to worry and won't need to run those commands.

```If you use yarn, you might want to delete `package-lock.json` file to avoid possible  resolution inconsistencies caused by unsynchronized lock files.```

This line should either be removed or we should fix why "resolution inconsistencies caused by unsynchronized lock files" happens. I'm not sure what it means

Comment (jakelumetta) [reply README.md:3]: Here's how I think the full README could read. The deploy to Vercel section is still TODO.

---------------

# Next.js + ButterCMS Starter Project
Live Demo: https://nextjs-starter-buttercms.vercel.app/

[Deploy with Vercel](https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2FButterCMS%2Fnextjs-starter-buttercms)


This Next.js starter project fully integrates with dynamic sample content from your ButterCMS account, including main menu, pages, blog posts, categories, and tags, all with a beautiful, custom theme with already-implemented search functionality. All of the included sample content is automatically created in your account dashboard when you sign up for a free trial of ButterCMS.

Once created, this project can be easily and quickly deployed to Vercel from the CLI ([see instructions below](#deploy-on-vercel))


## 1) Installation

First, install the dependencies by running one of these commands, depending on your current setup:
```bash
npm install # or yarn install
```

## 2) Set API Token

To fetch your ButterCMS content, add your API token as an environment variable.

```bash
$ echo 'NEXT_PUBLIC_BUTTER_CMS_API_KEY=<Your API Token>' >> .env
```

## 3) Run the local server

To view the app in the browser, you'll need to run the local development server:

```bash
npm run dev # or yarn dev
```

Congratulations! Your starter project is now live: [http://localhost:3000](http://localhost:3000).

## 4) Deploy

###  Vercel

The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.

1. Create a Vercel account at https://vercel.com/signup
2. Download [the CLI](https://vercel.com/download):

```bash
npm i -g vercel
```
3. Add the API key as a secret `vercel secrets add butter-cms-api-key "YOUR_API_TOKEN"`
4. Run `vercel` at the project root

#### Trigger Vercel builds on your ButterCMS content updates using webhooks

##### Create ButterCMS Webhook

...TODO...

Go to your Vercel account and Vercel deploy webhooks

Go to [https://buttercms.com/webhooks/](https://buttercms.com/webhooks/) and set up webhooks

Use of the ButterCMS webhooks to notify the application every time a post, page or collection item are added or modified, and reload the Next.js application.

...TODO...

## Preview mode
To see what application really look like before it’s published, add `PREVIEW=true` to your `.env` file and restart your server. From within your butter account, you can also preview changes to the app using our built-in iframe previewer. For more information, see [https://buttercms.com/kb/preview-urls](https://buttercms.com/kb/preview-urls).

**Review by jakelumetta — CHANGES_REQUESTED** [review]
@orlyohreally some initial feedback

File: lib/api.js (hunk 1 above)
Comment (orlyohreally) [reply lib/api.js:7]: @jakelumetta I'm not sure it's a good idea to commit `.env` file. I understand that it's just a starter app but if we want them to start off of this project we don't want them to spend time on removing .env file from their history and etc. We could add command to set `PREVIEW` in `.env` as a followup for command for setting `NEXT_PUBLIC_BUTTER_CMS_API_KEY `

File: pages/_document.js (hunk 2 above)
Comment (orlyohreally) [reply pages/_document.js:1]: To make sure that font-family is working correctly. I moved fort importing from `.css` file to head of the pages

File: README.md (hunk 3 above)
Comment (orlyohreally) [reply README.md:3]: > `` If you use yarn, you might want to delete `package-lock.json` file to avoid possible resolution inconsistencies caused by unsynchronized lock files. ``
> 
> This line should either be removed or we should fix why "resolution inconsistencies caused by unsynchronized lock files" happens. I'm not sure what it means

https://stackoverflow.com/a/46590427/11275812
I think since we deleted `yarn.lock` file from the repo, we could remove this message.


I have read your README version and I think it's good for our purposes.
